post_office.py

```diff
@@ -0,0 +1,81 @@
+class PostOffice:
+    """A Post Office class. Allows users to message each other.
+
+    :ivar int message_id: Incremental id of the last message sent.
+    :ivar dict boxes: Users' inboxes.
+
+    :param list usernames: Users for which we should create PO Boxes.
+    """
+
+    def __init__(self, usernames):
+        self.message_id = 0
+        self.boxes = {user: [] for user in usernames}
+
+    def send_message(self, sender, recipient, message_body, urgent=False):
+        """Send a message to a recipient.
+
+        :param str sender: The message sender's username.
+        :param str recipient: The message recipient's username.
+        :param str message_body: The body of the message.
+        :param urgent: The urgency of the message.
+        :type urgent: bool, optional
+        :return: The message ID, auto incremented number.
+        :rtype: int
+        :raises KeyError: if the recipient does not exist.
+        """
+        user_box = self.boxes[recipient]
+        self.message_id = self.message_id + 1
+        message_details = {
+            'id': self.message_id,
+            'body': message_body,
+            'sender': sender,
+        }
+        if urgent:
+            user_box.insert (0, message_details)
+        else:
+            user_box.append (message_details)
+        return self.message_id
+
+    def read_inbox(self, username : str, n : int = -1) -> list:
+        """
+        This function return the N  first messages in the user mail
+        :param str username: The message sender's username.
+        :param int n: N-th messages to be returned
+        :return lst: N-th messages
+        :rtype: list
+
+        """
+        return self.boxes[username][:n if n != -1 else len(self.boxes[username])]

```

Comment:
How are you dealing with messages that are already read. Please reread the assignment.